src/components/Cluster/ClusterDetail/AppDetailsModal/AppDetailsModal.js

```diff
@@ -0,0 +1,254 @@
+import { deleteApp, loadApps } from 'actions/appActions';
+import {
+  createAppConfig,
+  deleteAppConfig,
+  updateAppConfig,
+} from 'actions/appConfigActions';
+import {
+  createAppSecret,
+  deleteAppSecret,
+  updateAppSecret,
+} from 'actions/appSecretActions';
+import GenericModal from 'components/Modals/GenericModal';
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import Button from 'UI/Button';
+import ClusterIDLabel from 'UI/ClusterIDLabel';
+
+import EditChartVersionPane from './EditChartVersionPane';
+import InitialPane from './InitialPane';
+
+const modalPanes = {
+  deleteAppConfig: 'deleteAppConfig',
+  deleteAppSecret: 'deleteAppSecret',
+  deleteApp: 'deleteApp',
+  editChartVersion: 'editChartVersion',
+  initial: 'initial',
+};
+
+const AppDetailsModal = props => {
+  const [pane, setPane] = useState(modalPanes.initial);
+
+  if (props.app === null || typeof props.app === 'undefined') {
+    return <span />;
+  }
+
+  const appName = props.app.metadata.name;
+  const clusterId = props.clusterId;
+
+  function showPane(paneToShow) {
+    return function() {
+      setPane(paneToShow);
+    };
+  }
+
+  function onClose() {
+    showPane(modalPanes.initial)();
+    props.onClose();
+  }
+
+  async function loadAppsAndClose() {
+    await props.dispatch(loadApps(clusterId));
+    onClose();
+  }
+
+  async function _deleteAppConfig() {
```

Comment:
Why the `_` at the beginning of these functions' names? 😬

Reply:
They clash otherwise with the imported action names above.

Reply:
Can we rename them, maybe? Or if not, creating aliases for the actions by adding the `Action` suffix to them?
Ex.:

```javascript
import { deleteAppConfig as deleteAppConfigAction } from "..."
```